crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -108,6 +108,32 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.all().txs_iter(sender).last().map(|(_, tx)| Arc::clone(&tx.transaction))
     }
 
+    /// Returns the next valid transaction for the given sender.
+    pub(crate) fn get_highest_consecutive_transaction_by_sender(
+        &self,
+        sender: SenderId,
+        on_chain_nonce: u64,
+    ) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
+        let mut next_expected_nonce = on_chain_nonce + 1;
```

Comment:
i'm not sure this is entirely correct. if the on chain nonce for an account is 1, then the next transaction sent should have nonce 1, not nonce 2.